iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_id.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CXX_UNIQUE_ID_INL
+#define IOX_HOOFS_CXX_UNIQUE_ID_INL
+
+#include "iceoryx_hoofs/cxx/newtype.hpp"
+
+#include <atomic>
+
+namespace iox
+{
+namespace cxx
+{
+/// @brief Unique IDs within a process starting with 1. Monotonic increasing IDs are
+/// created with each call to the constructor. The IDs are copy/move constructible/assignable,
+/// comparable, sortable and convertable to the underlying value type.
+class UniqueId : public cxx::NewType<uint64_t,
+                                     cxx::newtype::ProtectedConstructByValueCopy,
+                                     cxx::newtype::Comparable,
+                                     cxx::newtype::Sortable,
+                                     cxx::newtype::Convertable,
+                                     cxx::newtype::CopyConstructable,
+                                     cxx::newtype::MoveConstructable,
+                                     cxx::newtype::CopyAssignable,
+                                     cxx::newtype::MoveAssignable>
+{
+  public:
+    using ThisType::ThisType;
+
+    /// @brief the constructor creates an ID which is greater than the previous created ID
+    UniqueId() noexcept;
+
+  private:
+    static std::atomic<value_type> g_IdCounter; // = 1U
```

Comment:
Please either remove the out commented code or initalise if possible:
```suggestion
    static std::atomic<value_type> g_IdCounter{1U}
```

Reply:
Seems like I missed the `static` :sweat_smile: A comment will do.

Reply:
nope, this must be done in the cpp. The comment was just to know what will be done in the `cpp`, similar to the `TypedUniqueId` class. But you are right, comments are not compiled and this should be removed or reworded to `// initialized in cpp file`